community/kernel/src/main/java/org/neo4j/kernel/DatabaseAvailability.java

```diff
@@ -63,18 +74,18 @@ public void stop()
     {
         // Database is no longer available for use
         // Deny beginning new transactions
-        availabilityGuard.deny(this);
+        availabilityGuard.deny( this );
 
         // Await transactions stopped
-        awaitNoTransactionsOr( 10_000  /* ms */);
+        awaitTransactionsClosedWithinTimeout();
     }
 
-    private void awaitNoTransactionsOr( int orUntilDeadline )
+    private void awaitTransactionsClosedWithinTimeout()
     {
-        long deadline = SYSTEM_CLOCK.currentTimeMillis() + orUntilDeadline;
-        while ( transactionMonitor.getNumberOfActiveTransactions() > 0 && SYSTEM_CLOCK.currentTimeMillis() < deadline)
+        long deadline = SYSTEM_CLOCK.currentTimeMillis() + awaitActiveTransactionDeadlineMillis;
+        while ( transactionMonitor.getNumberOfActiveTransactions() > 0 && SYSTEM_CLOCK.currentTimeMillis() < deadline )
         {
-            Thread.yield();
+            LockSupport.parkNanos( 10_000_1000 );
```

Comment:
The extra `1` in `1000` at the end makes the underscores not act as thousands dividers, this will park for `100.0001 ms`.

I'm sceptical about using `park`, since there is no matching `unpark`, what we really mean here is `sleep`, so why not use that method instead?